clang/include/clang/StaticAnalyzer/Core/PathSensitive/SVals.h

```diff
@@ -298,9 +299,12 @@ class SymbolVal : public NonLoc {
 /// Value representing integer constant.
 class ConcreteInt : public NonLoc {
 public:
-  explicit ConcreteInt(const llvm::APSInt &V) : NonLoc(ConcreteIntKind, &V) {}
+  explicit ConcreteInt(APSIntPtr V) : NonLoc(ConcreteIntKind, V.get()) {}
 
-  const llvm::APSInt &getValue() const { return *castDataAs<llvm::APSInt>(); }
+  APSIntPtr getValue() const {
+    // This is safe because in the ctor we take a safe APSIntPtr.
+    return APSIntPtr::unsafeConstructor(castDataAs<llvm::APSInt>());
```

Comment:
Is it possible to store `ApsIntPtr` in `NonLoc`?

Reply:
> Currently the infrastructure is set up for holding raw-pointers.

Could you elaborate on what preventing us from changing this?

Reply:
Basicall it all boils down to the fact that an sval has a single void pointer as a storage. I'm currently okay casting a pointer to and from void pointer of this storage. But I'd be uncomfortable bitcasting an APSIntPtr object into a void pointer storage, even if it only holds a single pointer inside. This is why I preferred unboxing the pointer and holding that inside the SVal.

Maybe I overlooked some other techniques. Let me know if you have something in mind.

Reply:
A union make this void pointer trickery marginally safer at the expense of declaring all the types we typepun this poor void storage pointer.
Maybe this is something to consider.

Reply:
According to [CppReference](https://en.cppreference.com/w/c/language/struct):
> A pointer to a struct can be cast to a pointer to its first member (or, if the member is a bit-field, to its allocation unit). Likewise, a pointer to the first member of a struct can be cast to a pointer to the enclosing struct.

Reply:
I see now. That was the source of the miscommunication here.